Upgrade .Net Core 3 SDK from preview5 to preview6 and related out of band Nuget packages from 2.1 to 3.0-preview6

bergmeister opened the pull request:
# PR Summary

Upgrade .Net Core 3 SDK from `preview5` to `preview6` and keep Nuget package `Microsoft.AspNetCore.Mvc.NewtonsoftJson` used for testing in sync with it

This required fixing the namespace ambiguity between `System.Management.Automation.AllowNullAttribute` and `System.Diagnostics.CodeAnalysis.AllowNullAttribute`

Please review the added files to the installer, do we want to ship all WPF localisations? Maybe yes for now but later it might be a good option to just include en-us and offer a cmdlet to download the other ones?

Fixes #9780 because it updates the `System.Management` NuGet package version from .Net 2.1 to 3.0-preview6

Updating also the 2 NuGet packages `System.ServiceProcess.ServiceController` and its dependency of `Microsoft.Windows.Compatibility` would've broken 2 tests and the reason for their failure seems to be unclear (i.e. whether it is a problem of PowerShell itself that is now being exposed or a regression in the `System.ServiceProcess.ServiceController`  NuGet package), therefore their update is deferred.

## PR Context

https://devblogs.microsoft.com/dotnet/announcing-net-core-3-0-preview-6

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): <!-- Experimental feature name(s) here -->
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: <!-- Number/link of that issue here -->
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

bergmeister committed:
Upgrade .Net Core 3 SDK from preview5 to preview6 and keep NuGet package Microsoft.AspNetCore.Mvc.NewtonsoftJson used for testing in sync with it

bergmeister committed:
Fix namespace ambiguity between 'System.Management.Automation.AllowNullAttribute' and 'System.Diagnostics.CodeAnalysis.AllowNullAttribute and cleanup some unused usings.

PoshChan commented:
@bergmeister, your last commit had 4 failures in `PowerShell-CI-windows`
TestCustomPipeNameCreationTooLongOnNonWindows
```powershell
Assert.True() Failure
Expected: True
Actual:   False
   at PSTests.Parallel.NamedPipeTests.TestCustomPipeNameCreationTooLongOnNonWindows() in D:\a\1\s\test\xUnit\csharp\test_NamedPipe.cs:line 46
```
TestCimInstanceProperty
```powershell
System.NotSupportedException : Specified method is not supported.
   at System.Management.Automation.Internal.PSDataCollectionReader`2.Read(Int32 count) in D:\a\1\s\src\System.Management.Automation\utils\ObjectReader.cs:line 520
   at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\pipelinebase.cs:line 416
   at System.Management.Automation.Runspaces.Pipeline.Invoke() in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\Pipeline.cs:line 531
   at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 5619
   at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 5466
   at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 4588
   at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 4783
   at System.Management.Automation.PowerShell.Invoke(IEnumerable input, PSInvocationSettings settings) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 2426
   at System.Management.Automation.PowerShell.Invoke() in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 2296
   at PSTests.Parallel.PSObjectTests.TestCimInstanceProperty() in D:\a\1\s\test\xUnit\csharp\test_PSObject.cs:line 120
```
TestRunspaceWithPowerShellAndInitialSessionState
```powershell
System.NotSupportedException : Specified method is not supported.
   at System.Management.Automation.Internal.PSDataCollectionReader`2.Read(Int32 count) in D:\a\1\s\src\System.Management.Automation\utils\ObjectReader.cs:line 520
   at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\pipelinebase.cs:line 416
   at System.Management.Automation.Runspaces.Pipeline.Invoke() in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\Pipeline.cs:line 531
   at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 5619
   at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 5466
   at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 4588
   at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 4783
   at System.Management.Automation.PowerShell.Invoke(IEnumerable input, PSInvocationSettings settings) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 2426
   at PSTests.Sequential.RunspaceTests.TestRunspaceWithPowerShellAndInitialSessionState() in D:\a\1\s\test\xUnit\csharp\test_Runspace.cs:line 86
```
TestRunspaceWithPowerShell
```powershell
System.NotSupportedException : Specified method is not supported.
   at System.Management.Automation.Internal.PSDataCollectionReader`2.Read(Int32 count) in D:\a\1\s\src\System.Management.Automation\utils\ObjectReader.cs:line 520
   at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\pipelinebase.cs:line 416
   at System.Management.Automation.Runspaces.Pipeline.Invoke() in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\Pipeline.cs:line 531
   at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 5619
   at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 5466
   at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 4588
   at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 4783
   at System.Management.Automation.PowerShell.Invoke(IEnumerable input, PSInvocationSettings settings) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 2426
   at PSTests.Sequential.RunspaceTests.TestRunspaceWithPowerShell() in D:\a\1\s\test\xUnit\csharp\test_Runspace.cs:line 55
```

bergmeister committed:
Fix MSI installer by adding the new DLLs

PoshChan commented:
@bergmeister, your last commit had 3 failures in `PowerShell-CI-windows`
TestCimInstanceProperty
```powershell
System.NotSupportedException : Specified method is not supported.
   at System.Management.Automation.Internal.PSDataCollectionReader`2.Read(Int32 count) in D:\a\1\s\src\System.Management.Automation\utils\ObjectReader.cs:line 520
   at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\pipelinebase.cs:line 416
   at System.Management.Automation.Runspaces.Pipeline.Invoke() in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\Pipeline.cs:line 531
   at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 5619
   at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 5466
   at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 4588
   at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 4783
   at System.Management.Automation.PowerShell.Invoke(IEnumerable input, PSInvocationSettings settings) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 2426
   at System.Management.Automation.PowerShell.Invoke() in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 2296
   at PSTests.Parallel.PSObjectTests.TestCimInstanceProperty() in D:\a\1\s\test\xUnit\csharp\test_PSObject.cs:line 120
```
TestRunspaceWithPowerShellAndInitialSessionState
```powershell
System.NotSupportedException : Specified method is not supported.
   at System.Management.Automation.Internal.PSDataCollectionReader`2.Read(Int32 count) in D:\a\1\s\src\System.Management.Automation\utils\ObjectReader.cs:line 520
   at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\pipelinebase.cs:line 416
   at System.Management.Automation.Runspaces.Pipeline.Invoke() in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\Pipeline.cs:line 531
   at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 5619
   at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 5466
   at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 4588
   at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 4783
   at System.Management.Automation.PowerShell.Invoke(IEnumerable input, PSInvocationSettings settings) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 2426
   at PSTests.Sequential.RunspaceTests.TestRunspaceWithPowerShellAndInitialSessionState() in D:\a\1\s\test\xUnit\csharp\test_Runspace.cs:line 86
```
TestRunspaceWithPowerShell
```powershell
System.NotSupportedException : Specified method is not supported.
   at System.Management.Automation.Internal.PSDataCollectionReader`2.Read(Int32 count) in D:\a\1\s\src\System.Management.Automation\utils\ObjectReader.cs:line 520
   at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\pipelinebase.cs:line 416
   at System.Management.Automation.Runspaces.Pipeline.Invoke() in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\Pipeline.cs:line 531
   at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 5619
   at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 5466
   at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 4588
   at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 4783
   at System.Management.Automation.PowerShell.Invoke(IEnumerable input, PSInvocationSettings settings) in D:\a\1\s\src\System.Management.Automation\engine\hostifaces\PowerShell.cs:line 2426
   at PSTests.Sequential.RunspaceTests.TestRunspaceWithPowerShell() in D:\a\1\s\test\xUnit\csharp\test_Runspace.cs:line 55
```

PoshChan commented:
@bergmeister, your last commit had 436 failures in `PowerShell-CI-macos`
(These are 5 of the failures)

InitialSessionState capacity.function capacity in initial session state should not be limited
```powershell
Exception calling "Invoke" with "0" argument(s): "Specified method is not supported."
at <ScriptBlock>, /Users/vsts/agent/2.152.1/work/1/s/test/powershell/engine/Api/InitialSessionState.Tests.ps1: line 31
31:         $ps.AddCommand('f4999').Invoke() | Should -Be "fn f4999"

```
InitialSessionState capacity.alias capacity in initial session state should not be limited
```powershell
Exception calling "Invoke" with "0" argument(s): "Specified method is not supported."
at <ScriptBlock>, /Users/vsts/agent/2.152.1/work/1/s/test/powershell/engine/Api/InitialSessionState.Tests.ps1: line 36
36:         $ps.AddCommand('a4999').Invoke() | Should -Be "fn f4999"

```
InitialSessionState capacity.variable capacity in initial session state should not be limited
```powershell
Exception calling "Invoke" with "0" argument(s): "Specified method is not supported."
at <ScriptBlock>, /Users/vsts/agent/2.152.1/work/1/s/test/powershell/engine/Api/InitialSessionState.Tests.ps1: line 41
41:         $ps.AddScript('$v4999').Invoke() | Should -Be "var v4999"

```
InitialSessionState capacity.function capacity should not be limited after runspace is opened
```powershell
Exception calling "Invoke" with "0" argument(s): "Specified method is not supported."
at <ScriptBlock>, /Users/vsts/agent/2.152.1/work/1/s/test/powershell/engine/Api/InitialSessionState.Tests.ps1: line 46
46:         $ps.AddScript('function f5000 { "in f5000" } f5000').Invoke() | Should -Be "in f5000"

```
InitialSessionState capacity.variable capacity should not be limited after runspace is opened
```powershell
Exception calling "Invoke" with "0" argument(s): "Specified method is not supported."
at <ScriptBlock>, /Users/vsts/agent/2.152.1/work/1/s/test/powershell/engine/Api/InitialSessionState.Tests.ps1: line 51
51:         $ps.AddScript('$v5000 = "var v5000"; $v5000').Invoke() | Should -Be "var v5000"

```

PoshChan commented:
@bergmeister, your last commit had 443 failures in `PowerShell-CI-linux`
(These are 5 of the failures)

InitialSessionState capacity.function capacity in initial session state should not be limited
```powershell
Exception calling "Invoke" with "0" argument(s): "Specified method is not supported."
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/engine/Api/InitialSessionState.Tests.ps1: line 31
31:         $ps.AddCommand('f4999').Invoke() | Should -Be "fn f4999"

```
InitialSessionState capacity.alias capacity in initial session state should not be limited
```powershell
Exception calling "Invoke" with "0" argument(s): "Specified method is not supported."
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/engine/Api/InitialSessionState.Tests.ps1: line 36
36:         $ps.AddCommand('a4999').Invoke() | Should -Be "fn f4999"

```
InitialSessionState capacity.variable capacity in initial session state should not be limited
```powershell
Exception calling "Invoke" with "0" argument(s): "Specified method is not supported."
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/engine/Api/InitialSessionState.Tests.ps1: line 41
41:         $ps.AddScript('$v4999').Invoke() | Should -Be "var v4999"

```
InitialSessionState capacity.function capacity should not be limited after runspace is opened
```powershell
Exception calling "Invoke" with "0" argument(s): "Specified method is not supported."
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/engine/Api/InitialSessionState.Tests.ps1: line 46
46:         $ps.AddScript('function f5000 { "in f5000" } f5000').Invoke() | Should -Be "in f5000"

```
InitialSessionState capacity.variable capacity should not be limited after runspace is opened
```powershell
Exception calling "Invoke" with "0" argument(s): "Specified method is not supported."
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/engine/Api/InitialSessionState.Tests.ps1: line 51
51:         $ps.AddScript('$v5000 = "var v5000"; $v5000').Invoke() | Should -Be "var v5000"

```

bergmeister commented:
@SteveL-MSFT It seems `preview6` of .Net Core 3 has bad regressions on non-Windows systems

SteveL-MSFT commented:
@bergmeister can you look in the inner exceptions to see what method it's referring to that is not supported?

bergmeister commented:
Yes, but only tomorrow.

iSazonov commented on src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddMember.cs:
```diff
@@ -11,6 +11,7 @@
 using System.Management.Automation.Host;
 using System.Management.Automation.Internal;
 using System.Reflection;
+using AllowNullAttribute = System.Management.Automation.AllowNullAttribute;
```
I think we can use new System.Diagnostics.CodeAnalysis.AllowNullAttribute

bergmeister replied:
Would there be an advantage, I guess we cannot remove the existing attribute due to it being public. Could this break edge cases where e.g. people analyze attributes on the AST?

SteveL-MSFT replied:
Let's not make this change in this PR

daxian-dbw replied:
I recommend to remove the unnecessary `[SuppressMessage("xxxx", "xxxxxx")]` from those affected files so that we can remove the `using System.Diagnostics.CodeAnalysis;` directive. Then `AllowNullAttribute` won't be ambiguous.

daxian-dbw replied:
@bergmeister Can you please address my comment? Thanks!

iSazonov commented on src/System.Management.Automation/engine/hostifaces/pipelinebase.cs:
```diff
@@ -413,7 +413,7 @@ public override Collection<PSObject> Invoke(IEnumerable input)
             // Execution completed successfully
             // 2004/06/30-JonN was ReadAll() which was non-blocking
-            return Output.NonBlockingRead(Int32.MaxValue);
+            return Output.Read(Int32.MaxValue);
```
Why do we change this?

bergmeister replied:
Sorry, this was a local change that I accidentally committed.

iSazonov commented on test/tools/WebListener/WebListener.csproj:
```diff
@@ -7,7 +7,7 @@
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.0.0-preview5-19227-01" />
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.0.0-preview6.19307.2" />
```
I think we should update all packages to latest versions. (As soon as we can compile the PR and pass tests.)

bergmeister replied:
Which other packages? This is the only package that I was aware of that is tightly coupled to the runtime

iSazonov replied:
All. Packages we use come from .Net Core 2.*. We need to update to latest preview 3.0.

bergmeister replied:
OK, thanks for the reference, I updated all those NuGet packages in a 2nd branch and ran the CI tests and there were only 2 failures on Windows, if you want I could add this update to this PR (especially since the packages are preview6 specific)? Here are the 2 failures:
https://christophbergmeister.visualstudio.com/PowerShell/_build/results?buildId=896&view=ms.vss-test-web.unified-test-results-tab

iSazonov replied:
I propose to update step by step (package by package).

bergmeister replied:
@iSazonov  I found the culprit that contains the breaking change that causes the 2 test failures:
` System.ServiceProcess.ServiceController`. Therefore we can probably update all NuGet packages except this one and `Microsoft.Windows.Compatibility` (because it depends on `System.ServiceProcess.ServiceController`). I am running builds at the moment in my fork to prove this theory. Should I also update the packages in `tool\packages\....` (that I think are used to create the NuGet packages for SMA, etc.)? Should the update of possible NuGet packages be in this PR or in a follow up PR (that I'd base on this PR?)?

bergmeister committed:
Revert accidental change in SMA for NonBlockingRead

bergmeister commented:
@SteveL-MSFT The failures were due to one line that I accidentally changed/commited.
Apart from `Download Artifacts` having a general issue in Azure DevOps at the moment, the Windows tests passed. On Linux, Mac there is only 1 failure in the `Pester No Sudo - CI` test suite:
```
Validate start of console host.No new assemblies are loaded

Expected exactly $null, but got @(@{InputObject=System.Security.Cryptography.Primitives.dll; SideIndicator=<=}, @{InputObject=System.Security.Principal.dll; SideIndicator=<=}).

at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Host/Startup.Tests.ps1: line 123
123:         $diffs | Should -BeExactly $null
```

SteveL-MSFT commented:
@bergmeister I believe you can update the startup tests to not expect those to be loaded.  Perhaps .NET did some refactoring that doesn't require those to be loaded during our startup usage.

bergmeister committed:
Fix Startup.Tests.ps1 expectations to expect System.Security.Cryptography.Primitives.dll and System.Security.Principal.dll to be only loaded on Windows

bergmeister commented:
Please review the added files to the installer, do we want to ship all WPF localisations? Maybe yes for now but later it might be a good option to just include en-us and offer a cmdlet to download the other ones?

iSazonov commented:
If MSFT team plans to include PowerShell 7.0 in Windows distribution the localization story becomes important. I think for preview we can put all in one package.

@bergmeister Thanks for the PR continue. Have you plans to update package versions to latest?

bergmeister commented:
Yes, see my comment here
https://github.com/PowerShell/PowerShell/pull/9888#discussion_r294045413

bergmeister committed:
Upgrade all NuGet packages except for System.ServiceProcess.ServiceController and Microsoft.Windows.Compatibility (which depends on it)

bergmeister committed:
Update web listener package as well

bergmeister committed:
Update references in NuGet packages as well to reflect changes

iSazonov commented:
@bergmeister You can continue in the PR if you want. Maintainers can add new commits as we did in past.

bergmeister committed:
Merge branch 'master' of https://github.com/PowerShell/PowerShell into netcore3.0-preview6

bergmeister committed:
Update System.ServiceProcess.ServiceController and Microsoft.Windows.Compatibility (which has a dependency on it). This will make 2 tests fail on Windows:
- Get-WinEvent.Get-WinEvent can retrieve events.Get-WinEvent can use the simplest of filters
- Set/New/Remove-Service cmdlet tests.Set-Service can accept a ServiceController as positional input

PoshChan commented on assets/files.wxs:
```diff
@@ -1649,6 +1649,9 @@
         <Component Id="cmp770700CA06564FB187D1CD2A79AA3DA8" Guid="{c244e7e1-3cb0-4880-aba5-e2be4fd78fbf}">
```
@bergmeister, your last commit had 2 failures in `PowerShell-CI-windows`
Set/New/Remove-Service cmdlet tests.Set-Service can accept a ServiceController as positional input
```powershell
Expected strings to be the same, but they were different.
String lengths are both 14.
Strings differ at index 0.
Expected: 'newdisplayname'
But was:  'testsetservice'
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Set-Service.Tests.ps1: line 321
321:             $service.DisplayName | Should -BeExactly $newdisplayname
```
Get-WinEvent.Get-WinEvent can retrieve events.Get-WinEvent can use the simplest of filters
```powershell
Cannot convert argument "value", with value: "System.Diagnostics.Eventing.Reader.EventLogRecord", for "TryAdd" to type "System.Diagnostics.Eventing.Reader.EventLogRecord": "Cannot convert the "System.Diagnostics.Eventing.Reader.EventLogRecord" value of type "System.Diagnostics.Eventing.Reader.EventLogRecord" to type "System.Diagnostics.Eventing.Reader.EventLogRecord"."
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Diagnostics\Get-WinEvent.Tests.ps1: line 61
61:                 $testEventDict.TryAdd($te.Id, $te)

```

bergmeister replied:
It is worthwhile noting that the first Windows service test throws an `ObjectDisposedException` before it fails:
```
		Set-Service : Cannot access a disposed object.
Object name: 'ServiceController'.
At C:\Users\cberg\git\PowerShell\test\powershell\Modules\Microsoft.PowerShell.Management\Set-Service.Tests.ps1:318 char:25
+ ...       $script = { Set-Service $service -DisplayName $newdisplayname }
+                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : ObjectNotFound: (:) [Set-Service], ObjectDisposedException
+ FullyQualifiedErrorId : InvalidOperationException,Microsoft.PowerShell.Commands.SetServiceCommand
```
![image](https://user-images.githubusercontent.com/9250262/59588305-31da2f00-90df-11e9-952d-be655ff6deae.png)

iSazonov replied:
@bergmeister Thanks for updating package versions!

Can you try to run Get-Service cmdlet manually in interactive session and see results?

bergmeister replied:
`Get-Service` on its own has no problem, it seems to be the interaction using `New-Service` and `Set-Service` that causes the failure later.

iSazonov replied:
It is expected behavior if we look line 2230 in Service.cs. It is not clear why we dispose ServiceController() before output.

iSazonov replied:
I do not found any changes in ServiceController.cs performed recently
https://github.com/dotnet/corefx/commits/60d8aa31a4c1ffa0e125bd7941aca171439a2f72/src/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceController.cs

It seems a root cause is in another place.
@SteveL-MSFT @daxian-dbw Have you any guesses?

PoshChan commented on assets/files.wxs:
```diff
@@ -1649,6 +1649,9 @@
         <Component Id="cmp770700CA06564FB187D1CD2A79AA3DA8" Guid="{c244e7e1-3cb0-4880-aba5-e2be4fd78fbf}">
```
@bergmeister, your last commit had 2 failures in `PowerShell-CI-windows`
Set/New/Remove-Service cmdlet tests.Set-Service can accept a ServiceController as positional input
```powershell
Expected strings to be the same, but they were different.
String lengths are both 14.
Strings differ at index 0.
Expected: 'newdisplayname'
But was:  'testsetservice'
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Set-Service.Tests.ps1: line 321
321:             $service.DisplayName | Should -BeExactly $newdisplayname
```
Get-WinEvent.Get-WinEvent can retrieve events.Get-WinEvent can use the simplest of filters
```powershell
Cannot convert argument "value", with value: "System.Diagnostics.Eventing.Reader.EventLogRecord", for "TryAdd" to type "System.Diagnostics.Eventing.Reader.EventLogRecord": "Cannot convert the "System.Diagnostics.Eventing.Reader.EventLogRecord" value of type "System.Diagnostics.Eventing.Reader.EventLogRecord" to type "System.Diagnostics.Eventing.Reader.EventLogRecord"."
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Diagnostics\Get-WinEvent.Tests.ps1: line 61
61:                 $testEventDict.TryAdd($te.Id, $te)

```

PoshChan replied:
@bergmeister, your last commit had 10 failures in `PowerShell-CI-macos`
(These are 5 of the failures)

float/double precision when converting to string.[double]-to-[string] conversion in PowerShell should use the precision specifier G15
```powershell
Expected strings to be the same, but they were different.
Expected length: 3
Actual length:   18
Strings differ at index 3.
Expected: '3.3'
But was:  '3.3000000000000003'
at <ScriptBlock>, /Users/vsts/agent/2.153.2/work/1/s/test/powershell/Language/Parser/Conversions.Tests.ps1: line 540
540:         $value -as [string] | Should -BeExactly $StringConversionResult
```
float/double precision when converting to string.[double]-to-[string] conversion in PowerShell should use the precision specifier G15
```powershell
Expected strings to be the same, but they were different.
Expected length: 3
Actual length:   18
Strings differ at index 3.
Expected: '6.6'
But was:  '6.6000000000000005'
at <ScriptBlock>, /Users/vsts/agent/2.153.2/work/1/s/test/powershell/Language/Parser/Conversions.Tests.ps1: line 540
540:         $value -as [string] | Should -BeExactly $StringConversionResult
```
float/double precision when converting to string.[double]-to-[string] conversion in PowerShell should use the precision specifier G15
```powershell
Expected strings to be the same, but they were different.
Expected length: 16
Actual length:   17
Strings differ at index 15.
Expected: '2.71828182845905'
But was:  '2.718281828459045'
at <ScriptBlock>, /Users/vsts/agent/2.153.2/work/1/s/test/powershell/Language/Parser/Conversions.Tests.ps1: line 540
540:         $value -as [string] | Should -BeExactly $StringConversionResult
```
float/double precision when converting to string.[double]-to-[string] conversion in PowerShell should use the precision specifier G15
```powershell
Expected strings to be the same, but they were different.
Expected length: 16
Actual length:   17
Strings differ at index 16.
Expected: '3.14159265358979'
But was:  '3.141592653589793'
at <ScriptBlock>, /Users/vsts/agent/2.153.2/work/1/s/test/powershell/Language/Parser/Conversions.Tests.ps1: line 540
540:         $value -as [string] | Should -BeExactly $StringConversionResult
```
float/double precision when converting to string.[float]-to-[string] conversion in PowerShell should use the precision specifier G7
```powershell
Expected strings to be the same, but they were different.
Expected length: 3
Actual length:   9
Strings differ at index 3.
Expected: '3.3'
But was:  '3.3000002'
at <ScriptBlock>, /Users/vsts/agent/2.153.2/work/1/s/test/powershell/Language/Parser/Conversions.Tests.ps1: line 540
540:         $value -as [string] | Should -BeExactly $StringConversionResult
```

adityapatwardhan commented:
@PoshChan please retry macOS

PoshChan commented:
@adityapatwardhan, successfully started retry of `PowerShell-CI-macOS`

SteveL-MSFT commented:
@bergmeister can you rebase off master and fix the conflict?  The macOS failure should be something that has already been fixed

bergmeister committed:
Merge branch 'master' of https://github.com/PowerShell/PowerShell into netcore3.0-preview6
Auto-resolved (git got confused because of the NJSon schema)
# Conflicts:
#	src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj

PoshChan commented on assets/files.wxs:
```diff
@@ -1646,6 +1646,9 @@
         <Component Id="cmp770700CA06564FB187D1CD2A79AA3DA8" Guid="{c244e7e1-3cb0-4880-aba5-e2be4fd78fbf}">
```
@bergmeister, your last commit had 2 failures in `PowerShell-CI-windows`
Set/New/Remove-Service cmdlet tests.Set-Service can accept a ServiceController as positional input
```powershell
Expected strings to be the same, but they were different.
String lengths are both 14.
Strings differ at index 0.
Expected: 'newdisplayname'
But was:  'testsetservice'
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Set-Service.Tests.ps1: line 321
321:             $service.DisplayName | Should -BeExactly $newdisplayname
```
Get-WinEvent.Get-WinEvent can retrieve events.Get-WinEvent can use the simplest of filters
```powershell
Cannot convert argument "value", with value: "System.Diagnostics.Eventing.Reader.EventLogRecord", for "TryAdd" to type "System.Diagnostics.Eventing.Reader.EventLogRecord": "Cannot convert the "System.Diagnostics.Eventing.Reader.EventLogRecord" value of type "System.Diagnostics.Eventing.Reader.EventLogRecord" to type "System.Diagnostics.Eventing.Reader.EventLogRecord"."
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Diagnostics\Get-WinEvent.Tests.ps1: line 61
61:                 $testEventDict.TryAdd($te.Id, $te)

```

PoshChan replied:
@bergmeister, your last commit had 2 failures in `PowerShell-CI-static-analysis`
Verify Markdown Links.Verify links in /home/vsts/work/1/s/docs/learning-powershell/powershell-beginners-guide.md.https://channel9.msdn.com/Series/GetStartedPowerShell3/06 should work
```powershell
retry of URL failed with error: 
at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 112
112:                                 throw "retry of URL failed with error: $($_.Message)"

```
Verify Markdown Links.Verify links in /home/vsts/work/1/s/docs/learning-powershell/powershell-beginners-guide.md.https://channel9.msdn.com/events/MMS/2012/SV-B406 should work
```powershell
retry of URL failed with error: 
at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 112
112:                                 throw "retry of URL failed with error: $($_.Message)"

```

PoshChan replied:
@bergmeister, your last commit had 2 failures in `PowerShell-CI-windows`
Set/New/Remove-Service cmdlet tests.Set-Service can accept a ServiceController as positional input
```powershell
Expected strings to be the same, but they were different.
String lengths are both 14.
Strings differ at index 0.
Expected: 'newdisplayname'
But was:  'testsetservice'
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Set-Service.Tests.ps1: line 321
321:             $service.DisplayName | Should -BeExactly $newdisplayname
```
Get-WinEvent.Get-WinEvent can retrieve events.Get-WinEvent can use the simplest of filters
```powershell
Cannot convert argument "value", with value: "System.Diagnostics.Eventing.Reader.EventLogRecord", for "TryAdd" to type "System.Diagnostics.Eventing.Reader.EventLogRecord": "Cannot convert the "System.Diagnostics.Eventing.Reader.EventLogRecord" value of type "System.Diagnostics.Eventing.Reader.EventLogRecord" to type "System.Diagnostics.Eventing.Reader.EventLogRecord"."
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Diagnostics\Get-WinEvent.Tests.ps1: line 61
61:                 $testEventDict.TryAdd($te.Id, $te)

```

bergmeister commented:
@SteveL-MSFT Done (the conflict could be auto-resolved, `git` was just confused due to the NJsonSchema update being one line below).
The 2 known failures have come back, that are due to the update of `System.ServiceProcess.ServiceController` (and `Microsoft.Windows.Compatibility`, which depends on it).  Please refer to the comment [here](https://github.com/PowerShell/PowerShell/pull/9888#discussion_r294177540) about some preliminary investigation

bergmeister committed:
Merge branch 'master' of https://github.com/PowerShell/PowerShell into netcore3.0-preview6
regenerate installer files after change in master, csproj conflict was just trivial newlines
# Conflicts:
#	assets/files.wxs
#	src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj

SteveL-MSFT commented:
@bergmeister if updating to newer versions of those assemblies is the cause of the failure, can we go back to an older one that works with Preview6?

bergmeister commented:
@SteveL-MSFT  Yes, I can just revert  b518c73a79b94942679da54d0385c6abd20ad60a (which means only the 2 Nuget packages `System.ServiceProcess.ServiceController` and its dependency of `Microsoft.Windows.Compatibility` are not updated but the rest is, i.e. the main framework and the majority of out of band NuGet packages)

iSazonov commented:
@SteveL-MSFT See my comment above https://github.com/PowerShell/PowerShell/pull/9888#discussion_r297610694

bergmeister committed:
Merge branch 'master' of https://github.com/PowerShell/PowerShell into netcore3.0-preview6

bergmeister committed:
Revert "Update System.ServiceProcess.ServiceController and Microsoft.Windows.Compatibility (which has a dependency on it). This will make 2 tests fail on Windows:"

This reverts commit b518c73a79b94942679da54d0385c6abd20ad60a.

bergmeister committed:
Fix installer by removing redundant System.Data.OleDb.dll entry

bergmeister commented:
@SteveL-MSFT I reverted  b518c73a79b94942679da54d0385c6abd20ad60a for a green build so that we can merge this in earlier and suggest to add a tracking issue for the 2 NuGet packages `System.ServiceProcess.ServiceController` and its dependency of `Microsoft.Windows.Compatibility` as they break 2 tests and the reason for their failure seems to be unclear (i.e. whether it is a problem of PowerShell itself that is now being exposed or a regression in the `System.ServiceProcess.ServiceController`  NuGet package)

iSazonov commented:
Should we report the problem with System.ServiceProcess.ServiceController to Core team?

daxian-dbw commented:
> Should we report the problem with System.ServiceProcess.ServiceController to Core team?

Yes we should report to the .NET team. Can we get a simple repro of the issue?

bergmeister committed:
Remove SuppressMessage attributes to make calls to AllowNullAttribute unique in terms of namespace to avoid having to add special static using statements as per code review.
Also remove one redundant using statement

bergmeister commented:
@daxian-dbw I applied the suggestion about the using statements from your comment [here](https://github.com/PowerShell/PowerShell/pull/9888#discussion_r299693943).
I m note sure if a repro is easy in terms of the usage of the APIs of the NuGet package, I cannot say if this is a bug in the NuGet package itself or something in PowerShell itself that only know starts to surface somehow. I'd appreciate it if you investigate this or we can open a tracking issue.

daxian-dbw reviewed (approved):
LGTM.
@SteveL-MSFT Are you OK with removing those outdated `SuppressMessage` attributes?

daxian-dbw commented:
@bergmeister Yes, please open a tracking issue with the information you have got. Thanks!

SteveL-MSFT commented:
@daxian-dbw yes, ok with removing the outdated attributes

bergmeister commented:
@daxian-dbw I opened issue #10067 to track the problem with the Nuget package. I will now resolve the merge conflict again. After this PR is merged, I plan to open a PR to demonstrate the failure so that the change is much smaller and we can focus on it and try experiments.

bergmeister committed:
Merge branch 'master' of https://github.com/PowerShell/PowerShell into netcore3.0-preview6
Re-hydrate files.wxs
# Conflicts:
#	assets/files.wxs
